app/models.py

```diff
@@ -1548,6 +1548,7 @@ def check_code(self, cde):
 NOTIFICATION_TECHNICAL_FAILURE = "technical-failure"
 NOTIFICATION_TEMPORARY_FAILURE = "temporary-failure"
 NOTIFICATION_PERMANENT_FAILURE = "permanent-failure"
+NOTIFICATION_PINPOINT_FAILURE = "pinpoint-failure"
```

Comment:
Let's name the status to untie from a specific technology. We could go back to AWS SES or we could support Twilio as well and other providers. Is there a way to name this without depending on the implementation (and extend the model on an abstract name)?

Reply:
Currently all the failures are SES related. This is now Piinpoint related and we wanted to store that and make it obv. I think as we grow and add more providers/ clients/ - we should add failures according to those implementations so we might know exactly where the failure came from and not have to traceback through logs

Reply:
I agree with not tying statuses to functionality or vendor specific terminology. That said, properly achieving vendor agnostic terminology for statuses, while still improving observability would take some effort. 

Since we're handling a conflict between a Request - `send to intl. #`, and the configuration of the target provider resource - `pinpoint not configured for intl. sending`, we _could_ use something more generic like:
- `PROVIDER_FAILURE`
- `PROVIDER_CONFLICT_FAILURE`
- `PROVIDER_VALIDATION_FAILURE`

Then capture the vendor specific info + failure reason in the `feedback_reason`. To Jumana's point - the current solution does provide added observability while requiring minimal changes to existing infrastructure for the time being.

Reply:
Hey I missed these replies. I am not big on tying our implementations with vendor specific names, especially when changing that to be generic would have been minimal effort IMO. We are starting a technical debt here that could have been avoided easily. This breaks abstractions and good design in general. I would bring this to the code review sessions.